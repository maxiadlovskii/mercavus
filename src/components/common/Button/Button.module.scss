@import '../../../styles/vars';

.Button {
  &.delete {
    border: 2px solid $alert;
    border-radius: 10px;
    cursor: pointer;
  }
}