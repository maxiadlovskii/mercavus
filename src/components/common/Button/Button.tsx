import React from 'react';
import {Button as ButtonInterface} from "../../../interfaces";
import styles from './Button.module.scss';
import classNames from 'classnames'
import {additionsClasses} from "../../../utils";

const Button: React.FC<ButtonInterface> =  ({
                                                text,
                                                additions = [],
                                                onClick,
                                                children
}) => {
    return <button
        className={classNames('Button', additionsClasses(additions, styles))}
        onClick={onClick}>{text || children}
    </button>
}

export default Button