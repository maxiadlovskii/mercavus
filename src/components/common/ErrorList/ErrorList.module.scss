@import "../../../styles/vars";
.list {
  color: $alert
}