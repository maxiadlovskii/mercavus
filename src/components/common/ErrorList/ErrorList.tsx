import React from 'react';
import { ErrorList as Interface } from '../../../interfaces'
import styles from './ErrorList.module.scss'

const ErrorList: React.FC<Interface> = ({ list }) => {
    return <ul className={styles['list']}>
        {list.map((error: string, i: number) => <li key={`error_${i}_${error}`}>{error}</li>)}
    </ul>
};

export default ErrorList