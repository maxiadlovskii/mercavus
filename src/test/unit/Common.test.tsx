import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure  } from 'enzyme';
configure({ adapter: new Adapter() });
import { shallowToJson } from 'enzyme-to-json';
import Button from '../../components/common/Button/Button';
import ErrorList from '../../components/common/ErrorList/ErrorList';
describe('Button', () => {
  it('should render correctly', () => {
    const output = shallow(
        <Button />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});

describe('Error List', () => {
  it('should render correctly', () => {
    const output = shallow(
        <ErrorList list={['ddd']}/>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});