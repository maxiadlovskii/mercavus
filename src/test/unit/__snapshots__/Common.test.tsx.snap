// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<button
  className="Button"
/>
`;

exports[`Error List should render correctly 1`] = `
<ul
  className="list"
>
  <li
    key="error_0_ddd"
  >
    ddd
  </li>
</ul>
`;
